### PR TITLE
Fix panic when scrolling through completion popup

### PR DESCRIPTION
**Pull request by midnightexigent**
Relevant issue: #1256 

Note that the documentation of [`Component::required_size`](https://github.com/helix-editor/helix/blob/master/helix-term/src/compositor.rs#L56) isn't clear enough as to who is supposed to handle the case where the sizes aren't in the bounds (parent or child ?) : 

On one hand:
> ... and the child should stay within those bounds

And on the other hand
> ... returned size might be larger than the viewport if the child is too big to fit. In this case the parent can use the values to calculate scroll.

In view of that, in this PR, the parent handles that case

### CODE REVIEW

**Comment by ZetaNumbers** [comment]
Also fixes #1262

**Review by ZetaNumbers — APPROVED** [review]

File: helix-term/src/ui/completion.rs [hunk 1]
```diff
@@ -328,8 +328,8 @@ impl Component for Completion {
                 let y = popup_y;
 
                 if let Some((rel_width, rel_height)) = markdown_doc.required_size((width, height)) {
-                    width = rel_width;
-                    height = rel_height;
+                    width = rel_width.clamp(0, width);
+                    height = rel_height.clamp(0, height);
```
Comment (archseer) [inline helix-term/src/ui/completion.rs:332]:
```suggestion
                    width = rel_width.min(width);
                    height = rel_height.min(height);
```

**Comment by archseer** [comment]
> PS: I am surprised i can "approve" this pr.

Anyone can approve a PR, it doesn't have any special behavior associated with it -- it's like a comment.

**Comment by archseer** [comment]
Thanks for the fix! 🎉

**Pull request merged**